base/strings/basic.jl

```diff
@@ -621,6 +621,28 @@ isascii(c::AbstractChar) = UInt32(c) < 0x80
     return 0 ≤ r < 0x80
 end
 
+@inline function _isascii(bytes::AbstractVector{UInt8},first,last)
+    n = last-first+1
+    nqword, epilog_bytes = divrem(n,8)
```

Comment:
```suggestion
    nqword, epilog_bytes = (n >> 3, n & 0x07) # divrem(n,8)
```
It doesn't look like LLVM simplifies `divrem` for the case where the divisor is a power of 2?

Reply:
that seems concerning. Is there something we can do on the Julia side to make it do better here?

Reply:
LLVM just can't prove `n > 0`

Reply:
That should be solveable by doing `n = (last - first + 1) % UInt` up top @ndinsmore